articles/android-rxjava/index.md

```diff
@@ -0,0 +1,127 @@
+* Android Studio installed
+* Basic understanding of Android development using Kotlin
+* Experience with retrofit and room
+
+#### What we'll look at :
+1. [Observables](#Observables)
+    * [observable](#observable)
+    * [flowable](#flowable)
+    * [single](#single)
+    * [completable](#completable)
+
+2. [Schedulers](#Schedulers)
+    * [newThread](#newThread)
+    * [io](#io)
+    * [computation](#computation)
+    * [mainThread](#mainThread)
+
+3. [Application in Android](#Application-in-Android)
+    * [retrofit](#retrofit)
+    * [room](#room)
+
+
+#### Observables
+RxJava, like the other ReactiveX libraries, uses the observable pattern. This pattern involves an `observer` who subscribes to an `observable`. The observer then receives data emitted by the observable and performs actions on it. An observable is an object or entity that emits data. We have five types of observers in RxJava, but we will look at only four. This is because the `Maybe` observable is not encouraged in Android applications. These observables have some standard methods in their classes to help in managing the emissions. The methods are `onNext`, `onCompleted`, `onSuccess` and `onError`.
+
+- **_onNext_** - the observable invokes this method when data when it receives data. The result contains data that has it has received. It is available in observable and flowable types since they can emit data more than once.
+
+- **_onSuccess_** - as the name states, it is called when an operation is successful. It is mostly used in `single` observables since they emit data only once.
+
+- **_onCompleted_** - observables call this method once it finishes emitting data or doing a task. It is available in all observable types.
+
+- **_onError_** - this is called when an error occurs. It emits a `Throwable` through which a developer can get the error details.
+
+
+##### `observable`
+This is the primary type of observable. It emits data as it receives it. It can be data from a collection like a list or a map. It has the `onNext`, `onError`, and `onCompleted` methods. It calls the onCompleted once onNext emits that last piece of data.
+
+##### `flowable`
+Flowable works the same way as an observable but with one extra capability. It is backpressure-aware. Sometimes the observable may emit data at a faster rate than the observer can consume. This can cause the application to crash or get an OOM (Out Of Memory) exception. Due to this, we use the flowable observable, which handles backpressure on behalf of the developer. A common way to get a backpressure issue is when loading a massive list of data from a local database.
+
+##### `single`
+This is another observable which emits data only once, as the name suggests. It does not have the onNext callback since it emits data once. It has the onSuccess instead. It emits the single piece of data in that callback. It is best used when making network calls or retrieving a single entity from a local database.
```

Comment:
I'd suggest you restructure the first sentence to "As the name suggests, this observable emits data only once". It is shorter and flows better.

Reply:
done...